docs/api-reference/extensions/collide-extension.md

```diff
@@ -0,0 +1,96 @@
+
+# CollideExtension
+
+The `CollideExtension` allows layers to hide features which overlap with other features. An example is a dense `ScatterplotLayer` with many points which overlap: by using this extension points that collide with others are hidden such that only one of the colliding points is shown. The collisions are computed on the GPU in realtime, allowing the collisions to be updated smoothly on every frame.
+
+To use this extension on a layer, add the `CollideExtension` to the layer's `extensions` prop.
+
+<!-- TODO: Codepen demo -->
+<div style="position:relative;height:450px"></div>
+
+```js
+import {ScatterplotLayer} from '@deck.gl/layers';
+import {CollideExtension} from '@deck.gl/extensions';
+
+const layer = new ScatterplotLayer({
+  id: 'points',
+  data: points,
+  extensions: [new CollideExtension()],
+  getPosition: d => d.COORDINATES,
+  getRadius: 10,
+  radiusUnits: 'pixels'
+})
+```
+
+## Installation
+
+To install the dependencies from NPM:
+
+```bash
+npm install deck.gl
+# or
+npm install @deck.gl/core @deck.gl/extensions
+```
+
+```js
+import {CollideExtension} from '@deck.gl/extensions';
+new CollideExtension();
+```
+
+To use pre-bundled scripts:
+
+```html
+<script src="https://unpkg.com/deck.gl@^8.9.0/dist.min.js"></script>
+<!-- or -->
+<script src="https://unpkg.com/@deck.gl/core@^8.9.0/dist.min.js"></script>
+<script src="https://unpkg.com/@deck.gl/extensions@^8.9.0/dist.min.js"></script>
+```
+
+```js
+new deck.CollideExtension();
+```
+
+## Constructor
+
+```js
+new CollideExtension();
+```
+
+## Layer Properties
+
+When added to a layer via the `extensions` prop, the `CollideExtension` adds the following properties to the layer:
+
+##### `collideEnabled` (Boolean, optional)
+
+Enable/disable collisions. If collisions are disabled, all objects are rendered. Defaults to `true`.
+
+##### `collideGroup` (string, optional)
+
+Collision group this layer belongs to. If it is not set, the 'default' collision group is used. Two (or more) layers that share the same `collideGroup` will be considered together when calculating collisions.
```

Comment:
I think a little more explanation is needed here. Maybe an example use case. My understanding is that you are only supposed to combine multiple layers into a single group if they use exactly the same data.